data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/stream/LogicalReplicationEventProcessor.java

```diff
@@ -333,4 +411,12 @@ private List<String> getPrimaryKeys(String schemaName, String tableName) {
 
         return progressState.getPrimaryKeyMap().get(databaseName + "." + schemaName + "." + tableName);
     }
+
+    private void handleMessageAndErrors(ByteBuffer message, Consumer<ByteBuffer> function, MessageType messageType) {
+        try {
+            eventProcessingTimer.record(() -> function.accept(message));
+        } catch (Exception e) {
+            LOG.error("Failed to process change event of type {}", messageType, e);
```

Comment:
Add metrics for this failure

Reply:
Is there a data loss here ? It is possible to reprocess this message ?

Reply:
I can add retries as well to handle transient issues. But if the message is somehow broken, there could be data loss here.

Reply:
Added retries in latest commit.